native/cocos/bindings/jswrapper/v8/Utils.cpp

```diff
@@ -152,9 +152,7 @@ void setReturnValueTemplate(const Value &data, const T &argv) {
     } else if (data.getType() == Value::Type::Number) {
         argv.GetReturnValue().Set(v8::Number::New(argv.GetIsolate(), data.toDouble()));
     } else if (data.getType() == Value::Type::BigInt) {
-        // Notice: Most return value of type `size_t` should be treated as Number.
-        // argv.GetReturnValue().Set(v8::BigInt::New(argv.GetIsolate(), data.toInt64()));
-        argv.GetReturnValue().Set(v8::Number::New(argv.GetIsolate(), static_cast<double>(data.toInt64())));
+        argv.GetReturnValue().Set(v8::BigInt::New(argv.GetIsolate(), data.toInt64()));
```

Comment:
How about defining a macro for switching to old behavior ?
```c++
#ifdef SE_INT64_VALUE_RETURN_DOUBLE
argv.GetReturnValue().Set(v8::Number::New(argv.GetIsolate(), static_cast<double>(data.toInt64())));
#else
argv.GetReturnValue().Set(v8::BigInt::New(argv.GetIsolate(), data.toInt64()));
#endif
```
And we need to notify developer that the logic has changed.